### PR TITLE
Fix progress bar rendering

### PR DESCRIPTION
**Pull request by scholtzan**
Fixes https://github.com/xi-editor/druid/issues/919

The problem causing this rendering issue is that insets get applied to a `Rect` with zero width. This results in a `Rect` of some negative width getting displayed when the progress bar is at zero.

### CODE REVIEW

File: druid/src/widget/progress_bar.rs [hunk 1]
```diff
@@ -86,18 +87,20 @@ impl Widget<f64> for ProgressBar {
         );
         ctx.fill(rounded_rect, &background_gradient);
 
-        //Paint the bar
+        // Paint the bar
         let calculated_bar_width = clamped * rounded_rect.width();
+
         let rounded_rect = Rect::from_origin_size(
-            Point::ORIGIN,
+            Point::new(-inset, 0.),
```
Comment (scholtzan) [inline druid/src/widget/progress_bar.rs:94]: We need to translate the rect to respect the x inset.

**Comment by xStrom** [comment]
Thanks for the fix!

Please also add an entry to [`CHANGELOG.md`](https://github.com/xi-editor/druid/blob/master/CHANGELOG.md) - [more instructions here](https://github.com/xi-editor/druid/blob/master/CONTRIBUTING.md).

**Comment by scholtzan** [comment]
Added an entry to the changelog

**Comment by cmyr** [comment]
Thanks @scholtzan!

**Pull request merged**